core/capabilities/ccip/ccipsolana/msghasher.go

```diff
@@ -79,20 +85,29 @@ func (h *MessageHasherV1) Hash(_ context.Context, msg cciptypes.Message) (ccipty
 		return [32]byte{}, fmt.Errorf("failed to decode extra args: %w", err)
 	}
 
-	var msgAccounts []solana.PublicKey
-	anyToSolanaMessage.ExtraArgs, msgAccounts, err = parseExtraArgsMapWithAccounts(extraDataDecodedMap)
+	ed, err := parseExtraDataMap(extraDataDecodedMap)
 	if err != nil {
 		return [32]byte{}, fmt.Errorf("failed to decode ExtraArgs: %w", err)
 	}
 
-	hash, err := ccip.HashAnyToSVMMessage(anyToSolanaMessage, msg.Header.OnRamp, msgAccounts)
+	anyToSolanaMessage.TokenReceiver = ed.tokenReceiver
+	anyToSolanaMessage.ExtraArgs = ed.extraArgs
+	accounts := ed.accounts
+	// if logical receiver is empty, don't prepend it to the accounts list
+	if !msg.Receiver.IsZeroOrEmpty() {
+		accounts = append([]solana.PublicKey{solana.PublicKeyFromBytes(msg.Receiver)}, accounts...)
```

Comment:
Ah I see, you're doing it here

Reply:
I'm not sure where this validation should live, but if the receiver is empty then the accounts list needs to be empty too. Maybe this already happens in CW (receiver execution is skipped?)

Reply:
Good point, we don't have that validation in CW. If logic receiver is empty, we just skip adding any accounts for messaging. We don't confirm that `extraArgs.Accounts` is empty

Reply:
I think that's the correct implementation though or we'll keep failing on these messages that were already committed. Validation should happen in the ccipSend logic: https://github.com/smartcontractkit/chainlink/blob/6f4282039edaedbd6165eef23e1cfba7a0cccb05/contracts/src/v0.8/ccip/FeeQuoter.sol#L1018-L1020

Reply:
If you're leaning towards this being in CW, I have another change pending that I can maybe include this with